mne/viz/epochs.py

```diff
@@ -65,71 +67,252 @@ def plot_epochs_image(epochs, picks=None, sigma=0., vmin=None,
+            picks = picks[:5]  # take 5 picks to prevent spawning many figs
+    else:
+        picks = np.atleast_1d(picks)
+
+    if ((fig is not None or axes is not None) and len(picks) > 1
+        and combine is None):
+        raise ValueError('Only single pick can be drawn to a figure/axis; '
+                         'provide only one pick, or use `combine`.')
 
     if set(units.keys()) != set(scalings.keys()):
         raise ValueError('Scalings and units must have the same keys.')
 
-    picks = np.atleast_1d(picks)
-    if (fig is not None or axes is not None) and len(picks) > 1:
-        raise ValueError('Only single pick can be drawn to a figure.')
+    ch_types = [channel_type(epochs.info, idx) for idx in picks]
+    if len(set(ch_types)) > 1 and groupby is None and combine is not None:
+        warn("Combining over multiple channel types. "
+             "Please use `groupby`.")
+    for ch_type in ch_types:
+        if ch_type not in scalings:
+            # We know it's not in either scalings or units since keys match
+            raise KeyError('%s type not in scalings and units' % ch_type)
+
+    if isinstance(axes, dict):
+        show = False
+
+    # First, we collect groupings of picks and types in two lists
+    # (all_picks, all_ch_types, names) -> groupby.
+    # Then, we construct a list of the corresponding data, names and evokeds
+    # (to_plot_list) -> combine.
+    # Then, we loop over this list and plot using _plot_epochs_image.
+
+    # groupby
+    all_picks, all_ch_types, names = _get_picks_and_types(
+        picks, ch_types, groupby, combine)
+    # all_picks is a list of lists of ints (picks); those lists will
+    # be length 1 if combine is None, else of length > 1.
+
+    # combine/construct list for plotting
+    to_plot_list = _get_to_plot(
+        epochs, combine, all_picks, all_ch_types, scalings, names)
+    # each entry of to_plot_list is: (data, ch_type, evoked, name)
+
+    # plot
+    figs = list()
+    for ii, (data_, ch_type, evoked, name) in enumerate(to_plot_list):
+        this_fig = _plot_epochs_image(
+            data_, sigma=sigma, vmin=vmin, vmax=vmax, colorbar=colorbar,
+            order=order, show=show, evoked=evoked, unit=units[ch_type],
+            scaling=scalings[ch_type], cmap=cmap, fig=fig,
+            axes=axes if not isinstance(axes, dict) else axes[name],
+            overlay_times=overlay_times, title=name, draw_evoked=True)
+        figs.append(this_fig)
+    return figs
+
+
+def _get_picks_and_types(picks, ch_types, groupby, combine):
+    "Helper for plot_epochs_image. Packs picks and types into a list."
+    if groupby is None:
+        if combine is not None:
+            picks = [picks]
+        return picks, ch_types, ch_types
+    else:
+        if groupby == "type":
+            all_picks, all_ch_types = list(), list()
+            for this_type in set(ch_types):
+                these_picks = picks[np.array(ch_types) == this_type]
```

Comment:
the remaining code would work if you construct the array only once above this loop: `ch_types = np.array(ch_types)`